command/polly_tts.js

```diff
@@ -326,6 +335,69 @@ var polly_tts = {
         }
       });
     });
+  },
+
+  /* 
+  * Takes an audio_url in S3:
+  * Gets the size, sample rate, duration, format, voice
+  * for the given file
+  *
+  * 1. Convert URL to local filename
+  * 2. Get the filesize.
+  * 3. Get the audio attributes using ffmpeg
+  *
+  * resolve: metadata object
+  * reject: err
+  */
+  getFileMetadata: function(audio_url) {
+    let audio_file = './' + audio_url.substr(audio_url.lastIndexOf('/') + 1);
+    return new Promise((resolve, reject) => {
+      // Get the filesize first:
+      let stats = fs.statSync(audio_file);
+      var fileSizeInBytes = stats['size'];
+      // Now get the audio attributes
+      try {
+        ffmpeg.ffprobe(audio_file, function(err, audioInfo) {
+          let metadata = {
+            format: 'mp3',
+            url: audio_url,
+            status: 'available',
+            voice: 'Joanna',
```

Comment:
Do you need to look up the voice from AudioFiles DB here?

Reply:
Good catch!  This is a bug.  I actually have the voice already... I just put Joanna in as a placeholder, but forgot to replace it with the actual voice.